src/Agent.Worker/Build/TrackingConfig.cs

```diff
@@ -60,6 +63,33 @@ public TrackingConfig()
             HashKey = hashKey;
             RepositoryUrl = repository.Url.AbsoluteUri;
             RepositoryType = repository.Type;
+            RepositoryDirectory = SourcesDirectory;
+            var path = repository.Properties.Get<string>(RepositoryPropertyNames.Path);
+            if (!string.IsNullOrEmpty(path))
+            {
+                if (path.IndexOfAny(Path.GetInvalidPathChars()) > -1)
+                {
+                    throw new InvalidOperationException($"{path} contains invalid path characters.");
+                }
+                else if (Path.GetFileName(path).IndexOfAny(Path.GetInvalidFileNameChars()) > -1)
+                {
+                    throw new InvalidOperationException($"{path} contains invalid folder name characters.");
+                }
+                else if (path.Contains($"..{Path.DirectorySeparatorChar}") || path.Contains($"..{Path.AltDirectorySeparatorChar}"))
```

Comment:
i think you need to split and look for any segment that is `..`

Reply:
it might be worthwhile to glance over `VSO\src\DistributedTask\Sdk\Server\SourceProviders\SourceProviderBase.cs`